### PR TITLE
Make measure understandable

### PR DESCRIPTION
**Pull request by MakaronKanon**
Split into smaller functions and add comments.

### CODE REVIEW

**Review by fabfors — APPROVED** [review]

File: src/Internal/Core.hs [hunk 1]
```diff
@@ -34,37 +35,81 @@ appendState (QState new) (QState state) = QState $ tensorVector new state
...
-
 -- | Vector state representations of qubits with 100% probaility
 -- to collapse to their bit counterparts
 newVector :: Bit -> QState
 newVector 0 = QState [1, 0]
 newVector 1 = QState [0, 1]
+
+
+-- Model probability as a rational number.
+type Prob = Rational
```
Comment (NicklasBoto) [inline src/Internal/Core.hs:46]: I'm not sure this is needed, but sure.

File: src/Internal/Core.hs [hunk 2]
```diff
@@ -34,37 +35,81 @@ appendState (QState new) (QState state) = QState $ tensorVector new state
...
+findQbitProb :: QBit -> QState -> (Prob, Prob)
+findQbitProb qbit qstate = (1 - prob1, prob1)
+  where
+    prob1 = findMarginProb1 qbit qstate
+
+-- | Finds the probability from all the amplitudes where that qubit is one.
+findMarginProb1 :: QBit -> QState -> Prob
+findMarginProb1 qbit qstate = sum $ map ampToProb (findMarginAmps1 qbit qstate)
+
+type Amplitude = Complex Double
```
Comment (NicklasBoto) [inline src/Internal/Core.hs:53]: Same here.

File: src/Internal/Core.hs [hunk 3]
```diff
@@ -34,37 +35,81 @@ appendState (QState new) (QState state) = QState $ tensorVector new state
...
 newVector 1 = QState [0, 1]
+
+
+-- Model probability as a rational number.
+type Prob = Rational
+
+-- | Given a qbit and the qstate, finds the probability of that qbit measuring to a 0 or 1.
+-- returns: (probToBeZero, probToBeOne)
+findQbitProb :: QBit -> QState -> (Prob, Prob)
+findQbitProb qbit qstate = (1 - prob1, prob1)
```
Comment (NicklasBoto) [inline src/Internal/Core.hs]: Kind of silly to have a function for this.
```suggestion
findQbitProb qbit qstate = (1 - prob1, prob1)
    where prob1 = sum $ map ampToProb (findMarginAmps1 qbit qstate)
```

File: src/Internal/Core.hs [hunk 4]
```diff
@@ -34,37 +35,81 @@ appendState (QState new) (QState state) = QState $ tensorVector new state
...
+  where
+    prob1 = findMarginProb1 qbit qstate
+
+-- | Finds the probability from all the amplitudes where that qubit is one.
+findMarginProb1 :: QBit -> QState -> Prob
```
Comment (NicklasBoto) [inline src/Internal/Core.hs]: Maybe move this to findQbitProb

**Review by NicklasBoto — CHANGES_REQUESTED** [review]
Looks good, but take a look at the suggestion.

File: src/Internal/Core.hs (hunk 2 above)
Comment (MakaronKanon) [reply src/Internal/Core.hs:53]: Leaving it for now

File: src/Internal/Core.hs (hunk 1 above)
Comment (MakaronKanon) [reply src/Internal/Core.hs:46]: Leaving it for now

File: src/Internal/Core.hs (hunk 3 above)
Comment (MakaronKanon) [reply src/Internal/Core.hs]: Fixed

File: src/Internal/Core.hs (hunk 4 above)
Comment (MakaronKanon) [reply src/Internal/Core.hs]: Fixed

**Pull request merged**